http/tus.go

```diff
@@ -0,0 +1,201 @@
+		mutex.Lock()
+		handler := th.getOrCreateTusHandler(d)
+		mutex.Unlock()
+
+		// Create upload directory for each request
+		uploadDir := filepath.Join(d.user.FullPath("/"), ".tmp_upload")
+		if err := os.MkdirAll(uploadDir, os.ModePerm); err != nil {
+			return http.StatusInternalServerError, err
+		}
+
+		switch r.Method {
+		case "POST":
+			handler.PostFile(w, r)
+		case "HEAD":
+			handler.HeadFile(w, r)
+		case "PATCH":
+			handler.PatchFile(w, r)
+		default:
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		}
+
+		// Isn't used
+		return 201, nil
+	})(w, r, &data{
+		store:    th.store,
+		settings: th.settings,
+		server:   th.server,
+	})
+
+	if err != nil {
+		http.Error(w, err.Error(), code)
+	} else if code >= 400 {
+		http.Error(w, "", code)
+	}
+}
+
+func (th tusHandler) createTusHandler(d *data) *tusd.UnroutedHandler {
+	uploadDir := filepath.Join(d.user.FullPath("/"), th.uploadDirName)
+	tusStore := filestore.FileStore{
+		Path: uploadDir,
+	}
+	composer := tusd.NewStoreComposer()
+	tusStore.UseIn(composer)
+
+	handler, err := tusd.NewUnroutedHandler(tusd.Config{
+		BasePath:              "/api/tus/",
+		StoreComposer:         composer,
+		NotifyCompleteUploads: true,
+	})
+	if err != nil {
+		panic(fmt.Errorf("Unable to create handler: %s", err))
+	}
+
+	// Goroutine to handle completed uploads
+	go func() {
+		for {
+			event := <-handler.CompleteUploads
+
+			if err := th.handleTusFileUploaded(handler, d, event); err != nil {
+				log.Printf("ERROR: couldn't handle completed upload: %s\n", err)
+			}
+		}
+	}()
+
+	return handler
+}
+
+func readMetadata(metadata tusd.MetaData, field string) (string, error) {
+	if value, ok := metadata[field]; ok {
+		return value, nil
+	} else {
+		return "", errors.New(fmt.Sprintf("Metadata field %s not found in upload request", field))
+	}
+}
+
+func (th tusHandler) handleTusFileUploaded(handler *tusd.UnroutedHandler, d *data, event tusd.HookEvent) error {
+	// Clean up only if an upload has been finalized
+	if !event.Upload.IsFinal {
+		return nil
+	}
+
+	filename, err := readMetadata(event.Upload.MetaData, "filename")
+	if err != nil {
+		return err
+	}
+	destination, err := readMetadata(event.Upload.MetaData, "destination")
+	if err != nil {
+		return err
+	}
+	overwrite, err := readMetadata(event.Upload.MetaData, "overwrite")
+	if err != nil {
+		return err
+	}
+	uploadDir := filepath.Join(d.user.FullPath("/"), th.uploadDirName)
+	uploadedFile := filepath.Join(uploadDir, event.Upload.ID)
+	fullDestination := filepath.Join(d.user.FullPath("/"), destination)
+
+	log.Printf("Upload of %s (%s) is finished. Moving file to destination (%s) "+
+		"and cleaning up temporary files.\n", filename, uploadedFile, fullDestination)
+
+	// Check if destination file already exists. If so, we require overwrite to be set
+	if _, err := os.Stat(fullDestination); !errors.Is(err, os.ErrNotExist) {
+		if overwrite, err := strconv.ParseBool(overwrite); err != nil {
```

Comment:
Maybe you can naming it as `overwriteStr`.
Cause `if overwrite, err := strconv.ParseBool(overwrite); err != nil {` would make it confused.
So `overwriteStr` = it's a string type of `overwrite` and `overwrite` is bool.

just my two cents :)
FYI: https://github.com/moby/moby/blob/master/daemon/logger/splunk/splunk.go#L176

Reply:
Fair point. Changed that as well.